### ISSUE TITLE
Add support for variable fonts on iOS/desktop

### ISSUE THREAD

**Issue by MikolajKakol** [JetBrains/compose-multiplatform#3127]
Currently, we cannot pass FontVariation arguments like we can on Android. We should be able to write code like this:
```
    Font( 
        R.font.roboto_slab, 
        weight = it, 
        variationSettings = FontVariation.Settings(FontVariation.weight(it.weight)) 
    ) 
```
Right now we cannot add those settings to modify custom axis. Like in https://speednet.pl/blog/fun-with-fonts2/

**Comment by pjBooms** [comment]
.. and the first part of the task is to provide common API for font loading from resources.

**Comment by RageshAntony** [comment]
@pjBooms 

So we can't use Fonts like this, we using in Android, in compose mutliplatform ?

```
val nunitoLight = FontFamily(Font(R.font.nunito_light)
val nunitoRegular = FontFamily(Font(R.font.nunito_regular))
val nunitoMedium = FontFamily(Font(R.font.nunito_medium))
val nunitoBold = FontFamily(Font(R.font.nunito_bold))
```

**Comment by RageshAntony** [comment]
@MikolajKakol 

The problem is, I put the font file in resources folder in commonMain, put couldn't access it like accessing in painterResource("res_file_name")

**Comment by MikolajKakol** [comment]
https://link.medium.com/AqBqFC0UDzb this is nice article about using fonts, I have almost identical implementation that worked for me

**Comment by Skaldebane** [comment]
Hi! Are there any plans to support this anytime soon?

The lack of this feature forces me to use static font files for Desktop/iOS, while using only one variable font file on Android. 

This makes some variable font weight/slant animations impossible to show anywhere other than Android.

I'd love to be able to use only the variable font files in the Shared module and define the variationAxis just as we do it in Android.

**Comment by vdcast** [comment]
> Hi! Are there any plans to support this anytime soon?
> 
> The lack of this feature forces me to use static font files for Desktop/iOS, while using only one variable font file on Android. This makes some variable font weight/slant animations impossible to show anywhere other than Android.
> 
> I'd love to be able to use only the variable font files in the Shared module and define the variationAxis just as we do it in Android.
> 
> What would it take to implement this? I can take on the Desktop side at least and implement support for it there!

Hi there! Which approach do you use to setup custom fonts in Compose Multiplatform, especially IOS?

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-3127">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.

**Comment by Skaldebane** [comment]
This was fixed, now available in Compose 1.8.0-alpha01!
